core/src/main/java/hudson/model/Queue.java

```diff
@@ -1464,7 +1470,7 @@ public void maintain() {
                     Collections.sort(blockedItems, QueueSorter.DEFAULT_BLOCKED_ITEM_COMPARATOR);
                 }
                 for (BlockedItem p : blockedItems) {
-                    String taskDisplayName = p.task.getFullDisplayName();
+                    String taskDisplayName = LOGGER.isLoggable(Level.FINEST) ? p.task.getFullDisplayName() : null;
```

Comment:
I recommended @christ66 do this in #1863 but he did not. Turns out this had a far worse consequence than wasting a bit of CPU time, which will become clearer in another PR.